src/gdbstub_impl/ext/reverse_exec.rs

```diff
@@ -0,0 +1,106 @@
+                    Ok(Some(0x03)) => true, // 0x03 is the interrupt byte
+                    Ok(Some(_)) => false,   // it's nothing that can't wait...
+                    Ok(None) => false,
+                    Err(e) => {
+                        err = Err(Error::ConnectionRead(e));
+                        true // break ASAP if a connection error occurred
+                    }
+                };
+
+                let stop_reason = match target.base_ops() {
+                    BaseOps::MultiThread(ops) => {
+                        if let Some(ops) = ops.support_reverse_cont() {
+                            ops.reverse_cont(&mut check_gdb_interrupt)
+                                .map_err(Error::TargetError)?
+                        } else {
+                            return Err(Error::PacketUnexpected);
+                        }
+                    }
+
+                    BaseOps::SingleThread(_ops) => {
+                        return Err(Error::PacketUnexpected);
+                    }
+                };
+
+                err?;
+
+                // FIXME: Unnecessary unwrap
+                self.finish_exec(res, target, stop_reason)?.unwrap()
+            }
+        };
+
+        Ok(handler_status)
+    }
+
+    // FIXME: De-duplicate with above code?
+    pub(crate) fn handle_reverse_step(
+        &mut self,
+        res: &mut ResponseWriter<C>,
+        target: &mut T,
+        command: ReverseStep,
+    ) -> Result<HandlerStatus, Error<T::Error, C::Error>> {
+        let handler_status = match command {
+            ReverseStep::bs(_) => {
+                let tid = match self.current_resume_tid {
+                    // N.B: Can't single-step all cores.
+                    SpecificIdKind::All => return Err(Error::PacketUnexpected),
+                    SpecificIdKind::WithId(tid) => tid,
+                };
+
+                // FIXME: This block is duplicated from the vCont code.
+                let mut err = Ok(());
+                let mut check_gdb_interrupt = || match res.as_conn().peek() {
+                    Ok(Some(0x03)) => true, // 0x03 is the interrupt byte
+                    Ok(Some(_)) => false,   // it's nothing that can't wait...
+                    Ok(None) => false,
+                    Err(e) => {
+                        err = Err(Error::ConnectionRead(e));
+                        true // break ASAP if a connection error occurred
+                    }
+                };
+
+                let stop_reason = match target.base_ops() {
+                    BaseOps::MultiThread(ops) => {
+                        if let Some(ops) = ops.support_reverse_step() {
+                            ops.reverse_step(tid, &mut check_gdb_interrupt)
+                                .map_err(Error::TargetError)?
+                        } else {
+                            return Err(Error::PacketUnexpected);
+                        }
```

Comment:
Instead of doing this "late" operation check, you should lift this up to the top of the function, matching how other extensions handle this: https://github.com/daniel5151/gdbstub/blob/3084b311c20a41d966271be023ea39c13870a9cf/src/gdbstub_impl/ext/monitor_cmd.rs#L13-L16

One minor road-bump I forsee is that `let ops` will have to be an enum of either the single threaded or multi threaded ops. You'll probably have to manually define a locally scoped enum similar to the following:

```rust
enum ReverseStepOps { 
    SingleThread(<single threaded ops type>), 
    MultiThread(<multi threaded ops type>) 
}
```

Reply:
Oh, and I should mention that this isn't just a style nit, and has a tangible binary size impact.

The compiler is smart enough to inline `handle_X` methods (since they are only ever called once), and by having the early ops guard, the optimizer is able to completely discard the body of the function in cases where the target hasn't implemented the IDET.

Reply:
Really? Huh, very interesting. I'm toying around with the `__dead_code_marker` stuff now (which is pretty nifty), and it still looks like Rust isn't eliminating my new handlers for `bc` and `bs`.
I'll push the new iteration in a bit so you can take a look.

Reply:
Yeah, I've been ricing `gdbstub` pretty hard, often leaning on godbolt and `cargo asm` to make sure things are in good shape.

I'd also recommend running `example_no_std/check_size.sh` to make sure you're not accidentally ballooning the base-configuration code size.